### PR TITLE
FOGL-7023 set PYTHONPATH

### PR DESCRIPTION
**Pull request by MarkRiddoch**
I've merged your FOG-6180.v2 into this branch so that I can get the tests run by Jenkins. 

**NB** Using this review as a way to get Jenkins to run the tests not as a real review

### CODE REVIEW

**Comment by Mohit04tomar** [comment]
Out of 200 iterations of test, on very first iteration 3 test cases failed:

```
Repeating all tests (iteration 1) . . .

Note: Randomizing tests' orders with a seed of 63532 .
[==========] Running 9 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 9 tests from MQTTScripted
[ RUN      ] MQTTScripted.UpdateScript
[       OK ] MQTTScripted.UpdateScript (34 ms)
[ RUN      ] MQTTScripted.NoFile
[       OK ] MQTTScripted.NoFile (0 ms)
[ RUN      ] MQTTScripted.RuntimError
/home/ubuntu/workspace/oglamp-south-mqtt-scripted_PR-45/tests/test_python.cpp:111: Failure
      Expected: python.setScript(fname)
      Which is: false
To be equal to: true
[  FAILED  ] MQTTScripted.RuntimError (1 ms)
[ RUN      ] MQTTScripted.NoDict
/home/ubuntu/workspace/oglamp-south-mqtt-scripted_PR-45/tests/test_python.cpp:80: Failure
      Expected: python.setScript(fname)
      Which is: false
To be equal to: true
[  FAILED  ] MQTTScripted.NoDict (0 ms)
[ RUN      ] MQTTScripted.NoFunc
[       OK ] MQTTScripted.NoFunc (0 ms)
[ RUN      ] MQTTScripted.RuntimError2
[       OK ] MQTTScripted.RuntimError2 (1 ms)
[ RUN      ] MQTTScripted.PluginInfoConfigParse
[       OK ] MQTTScripted.PluginInfoConfigParse (0 ms)
[ RUN      ] MQTTScripted.SimplePython
/home/ubuntu/workspace/oglamp-south-mqtt-scripted_PR-45/tests/test_python.cpp:21: Failure
      Expected: python.setScript(fname)
      Which is: false
To be equal to: true
[  FAILED  ] MQTTScripted.SimplePython (0 ms)
[ RUN      ] MQTTScripted.PluginInfo
[       OK ] MQTTScripted.PluginInfo (0 ms)
[----------] 9 tests from MQTTScripted (36 ms total)

[----------] Global test environment tear-down
[==========] 9 tests from 1 test case ran. (36 ms total)
[  PASSED  ] 6 tests.
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] MQTTScripted.RuntimError
[  FAILED  ] MQTTScripted.NoDict
[  FAILED  ] MQTTScripted.SimplePython

 3 FAILED TESTS

```
For more details, please check: http://jenkins.dianomic.com:8080/job/dianomic-plugins-tester/job/foglamp-south-mqtt-scripted/view/change-requests/job/PR-45/

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                currentBuild.result = 'SUCCESS'
+                echo "Failed to build FogLAMP; required to run the tests!"
+                return
+            }
+
+            try {
+                stage("Run Tests") {
+                    echo "Executing tests..."
+                    sh '''
+                        export FOGLAMP_ROOT=${WORKSPACE}/FogLAMP && export PYTHONPATH=/home/foglamp/FogLAMP/python:
```
Comment (ashish-jabble) [inline Jenkinsfile:35]: PYTHONPATH value is wrong here. It should be from ${WORKSPACE}/FogLAMP

File: tests/main.cpp [hunk 2]
```diff
@@ -2,15 +2,18 @@
...
 
 using namespace std;
 
 int main(int argc, char **argv) {
     testing::InitGoogleTest(&argc, argv);
 
-    testing::GTEST_FLAG(repeat) = 200;
-    testing::GTEST_FLAG(shuffle) = true;
-    testing::GTEST_FLAG(death_test_style) = "threadsafe";
+    setenv("PYTHONPATH", ".", 1);
```
Comment (ashish-jabble) [inline tests/main.cpp:13]: Is this a workaround?

Comment (MarkRiddoch) [reply tests/main.cpp:13]: Yes, but maybe if the other file is corrected it is not needed.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                currentBuild.result = 'SUCCESS'
+                echo "Failed to build FogLAMP; required to run the tests!"
+                return
+            }
+
+            try {
+                stage("Run Tests") {
+                    echo "Executing tests..."
+                    sh '''
+                        export FOGLAMP_ROOT=${WORKSPACE}/FogLAMP && export PYTHONPATH=${WORKSPACE}/FogLAMP:
```
Comment (ashish-jabble) [inline Jenkinsfile:35]: I mean `export PYTHONPATH=${WORKSPACE}/FogLAMP/python`

**Pull request closed without merging**